src/EventFeed.Producer.Caching/CachingEventFeedReader.fs

```diff
@@ -0,0 +1,36 @@
+﻿namespace EventFeed.Producer.Caching
+
+open EventFeed
+open EventFeed.Producer.Abstractions
+open Microsoft.Extensions.Caching.Memory
+open System
+
+/// Provides an in-memory caching layer that wraps an IEventFeedReader
+type CachingEventFeedReader(underlyingReader : IEventFeedReader, cache : IMemoryCache) =
+    static let completePageExpirationTime = TimeSpan.FromMinutes(5)
+    static let lastPageExpirationTime = TimeSpan.FromSeconds(1)
+    
+    let isLastPage (pageNumber : int) =
+        let eventNumbers = underlyingReader.EventNumbers()
+        let totalPages = Paging.totalPages eventNumbers.EventCount eventNumbers.EventsPerPage
+        pageNumber = totalPages
+
+    let getCacheEntryOptions (pageNumber : int) =
+        match isLastPage(pageNumber) with
+        | true ->  new MemoryCacheEntryOptions(AbsoluteExpirationRelativeToNow = lastPageExpirationTime)
+        // Would be nice if we can use Size here so completed pages only get evicted when cache reaches max size
+        | false -> new MemoryCacheEntryOptions(AbsoluteExpirationRelativeToNow = completePageExpirationTime )
+
+    interface IEventFeedReader with
+
+        member this.EventNumbers() = underlyingReader.EventNumbers()
+
+        member this.ReadPage pageNumber =
+            match cache.Get(pageNumber) with
```

Comment:
`Get` has a `Get<T>` overload right? Could use that and not have to do the type check?
What may be more idiomatic but have a slight perf hit so wouldn't worry about it is having a closure over `cache` that returns an `Option` and then match on `Some` and `None`.
Or we can do it and tune later when we have benchmarks :)

Reply:
Yes, you're right! I was looking for that but thought I was mistaken, but for some reason these extension methods are not showing up in my IntelliSense in F#. I also don't get auto import, or rename refactoring support. Is that normal or my setup?